enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,438 @@
+number of active pods: 1 in normal operation and 2 when migration is in progress.
+This means the PDB is always at limit. This has two main functions.
+
+1. As we described. KubeVirt is doing a custom eviction, so it lets the eviction
+  request continue to the eviction API where it fails because the PDB is at
+  limit and has `disruptionsAllowed` set to 0. This blocks the node drain in
+  the short term as it cannot evict the virtual machine pods. But once the
+  migration completes after some time, it will stop the original pod, so the
+  drain can progress in the end.
+
+2. They want the pods to be protected even when the webhook is down for any
+  reason. The API initiated eviction will always respect the PDB and let the VM
+  pod live. In that case this would block the node drain.
+
+There does not seem to be another option how they would achieve this
+(for example by tinkering with PDBs).
+
+In OpenShift we are alerting with warning severity on any PDB that is at limit
+([PodDisruptionBudgetAtLimit][3]). Which means we check if a PDB status is
+`current_healthy == desired_health` (and `expected_pods > 0`). Normal workloads
+should have at least one more pod than is desired to support eviction.
+Reporting these alerts is important because it helps users to pinpoint the
+workloads that are violating this and could block node drain.
+This is important for node maintenance and [cluster upgrades][4].
+
+KubeVirt on the other hand uses PDBs, but is going around the intended use case
+of eviction. They do this in order to support their functionality, but the
+eviction or PDBs were never designed with such use cases in mind.
+KubeVirt use case could be explored in a KEP, but that is a different topic.
+
+Users that wish to use Live Migration feature then trigger this alert. From
+their perspective they are just using KubeVirt API as intended and this warning
+is unnecessary. The warning is a way to tell people that there are not enough
+replicas in their workload  and that the node drain might not work.
+For KubeVirt that is not the case - the intended number of pods per VM is 1
+and the eviction is handled in a custom way.
+
+For users with live migratable VMs, this alert is a constant occurrence.
+Apart from the alert being annoying, this diminishes the value of the alert,
+and it could be very easily overlooked when other workloads run into the PDB
+limit.
+
+### User Stories
+
+Components should be able to mark certain resources in the cluster that should
+not trigger specific alerts. However, this should only be allowed with the
+consent of the alert owner.
+
+
+* As an OpenShift component owner I want to be able to have a mechanism to opt
+  out of alerts, where it makes sense.
+
+* As a KubeVirt user I do not want to see alerts for PDBs that are guarding
+  virtual machines with `LiveMigrate` eviction strategy.
+
+### Goals
+
+- Introduce a generic CRD for alert exclusion that can be used by the
+  ValidatingAdmissionPolicy.
```

Comment:
+1 in AdmPolicy.
My only worry is that it's only in alpha in 1.26.

Well, maybe a good use-case to get it to beta ...

Reply:
we could also start with tech preview first to see it works for us, before we enable it by default